internal/services/azapi_resource.go

```diff
@@ -502,6 +502,15 @@ func (r *AzapiResource) CreateUpdate(ctx context.Context, requestPlan tfsdk.Plan
 		defer locks.UnlockByID(lockId)
 	}
 
+	// preflight validation: currently only supports the subscription scope validation
+	if r.ProviderData.Features.EnablePreflight && isNewResource && strings.Contains(strings.ToLower(plan.ParentID.ValueString()), "/subscriptions/") {
```

Comment:
Could we use strings.HasPrefix() here as the `/subscriptions/` must always be at the start?

Reply:
updated. Thanks.